userspace/libsinsp/sinsp.cpp

```
@@ -486,6 +475,18 @@ void sinsp::open_common(scap_open_args* oargs,
 		throw scap_open_exception(error, scap_rc);
 	}
 
+	// Validate the schema version of the plugins in case the engine provides the event schema
+	// version it complies with
+	if(vtable && vtable->get_schema_version) {
```

This means that the check is not performed for the engines that do not implement, `get_schema_version`

---

btw. there is already ASSERT for vtable != NULL in `scap_init()` but always nicer not to depend on external checks...

---

good point, I prefer removing the null check on `vtable` because we have enough defensive code in the repo and it tends to make things harder when unexpected things happen. Better to fail fast.